internal/serialization/generic_compact.go

```diff
@@ -19,109 +14,74 @@ type SchemaInfo struct {
 	TypeName string
 }
 
-type GenericCompactDeserializer struct {
-	SchemaInfos *sync.Map
-}
-
-func NewGenericCompactDeserializer() *GenericCompactDeserializer {
-	return &GenericCompactDeserializer{
-		SchemaInfos: &sync.Map{},
-	}
-}
+type GenericCompactDeserializer struct{}
 
-func (cm *GenericCompactDeserializer) Read(schema *hazelcast.Schema, reader serialization.CompactReader) interface{} {
-	v := reflect.New(cm.makeType(schema))
-	for i, fd := range schema.FieldDefinitions() {
+func (cm GenericCompactDeserializer) Read(schema *hazelcast.Schema, reader serialization.CompactReader) interface{} {
+	fds := schema.FieldDefinitions()
+	cs := make(ColumnList, len(fds))
+	for i, fd := range fds {
 		r := compactReaders[fd.Kind]
-		value := reflect.ValueOf(r(reader, fd.Name))
-		if value.Interface() == nil {
-			continue
+		v := r(reader, fd.Name)
+		c := Column{
+			Name:  fd.Name,
+			Value: v,
 		}
-		v.Elem().Field(i).Set(value)
-	}
-	return v.Interface()
-}
-
-func (cm *GenericCompactDeserializer) makeType(schema *hazelcast.Schema) reflect.Type {
-	var t reflect.Type
-	vi, ok := cm.SchemaInfos.Load(schema.ID())
-	if ok {
-		v := vi.(SchemaInfo)
-		t = v.Type
-	} else {
-		fds := schema.FieldDefinitions()
-		fs := make([]reflect.StructField, len(fds))
-		for i, f := range fds {
-			fs[i] = reflect.StructField{
-				Name: fmt.Sprintf("Field%03d", i),
-				Type: fieldKindToType[f.Kind],
-				Tag:  reflect.StructTag(fmt.Sprintf("json:\"%s\"", f.Name)),
-			}
+		if v == nil {
+			c.Type = TypeNil
```

Comment:
We already have `FieldKindNotAvailable` type, isn't this available through `fd.kind`?